src/main/java/org/folio/marc4ld/service/marc2ld/bib/MarcBib2LdMapperImpl.java

```diff
@@ -49,14 +54,41 @@ public class MarcBib2LdMapperImpl implements MarcBib2ldMapper {
   private final EmptyEdgesCleaner emptyEdgesCleaner;
 
   @Override
-  public Resource fromMarcJson(String marc) {
+  public Optional<Resource> fromMarcJson(String marc) {
     if (isEmpty(marc)) {
       log.warn("Given marc is empty [{}]", marc);
-      return null;
+      return Optional.empty();
+    }
+    var records = marcReaderProcessor.readMarc(marc).filter(this::isMonograph).toList();
+    if (records.isEmpty()) {
+      log.warn("Given marc is not monograph, skipping: [{}]", marc);
+      return Optional.empty();
+    }
+    return Optional.of(createInstanceAndWorkResource(records));
+  }
+
+  private boolean isMonograph(Record record) {
+    var leader = record.getLeader();
+    if (leader == null) {
```

Comment:
Minor: It's a bit more readable to use Objects.isNull with static import

Reply:
fixed.